sklearn/utils/tests/test_validation.py

```diff
@@ -1213,3 +1214,62 @@ def test_check_sparse_pandas_sp_format(sp_format):
     assert sp.issparse(result)
     assert result.format == sp_format
     assert_allclose_dense_sparse(sp_mat, result)
+
+
+@pytest.mark.parametrize('ntype1, ntype2', [
+    ("longdouble", "float16"),
+    ("float16", "float32"),
+    ("float32", "double"),
+    ("int16", "int32"),
+    ("int32", "long"),
+    ("byte", "uint16"),
+    ("ushort", "uint32"),
+    ("uint32", "uint64"),
+    ("uint8", "int8"),
+])
+def test_check_pandas_sparse_invalid(ntype1, ntype2):
+    pd = pytest.importorskip("pandas", minversion="0.25.0")
+    if parse_version(pd.__version__) >= parse_version('1.1'):
+        pytest.skip("issue of generating an object dtype coo_matrix"
+                    " from a DataFrame with extension arrays with"
+                    " different numeric types was fixed in pandas"
+                    " 1.1.0")
```

Comment:
Should we have a test that make sure this works for `minversion=1.1.0`?

Reply:
Right. makes a lot of sense. done.